src/main/java/ua/com/danit/service/CommentService.java

```diff
@@ -26,13 +25,9 @@ public CommentService(CommentRepository commentRepository, UserService userServi
     return commentRepository.findById(commentId);
   }
 
-  public Comment createComment(Comment comment) {
-    Post post = postService.getCurrentPost();
-    User author = userService.getCurrentUser();
-
-    comment.setPost(post);
-    comment.setAuthor(author);
-
+  public Comment createComment(Comment comment, Long postId) {
+    comment.setPost(postService.getPostById(postId));
+    comment.setAuthor(userService.getCurrentUser());
```

Comment:
let's explicitly set id to null for create-like method. If this method will be called outside, anyone can pass comment entity with existing id. In this case entity update will be performed

Reply:
CommentRequest has only "text" field